bitrise.yml

```diff
@@ -3,51 +3,221 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
 
 app:
   envs:
-  - STEP_VERSION: 3.1.4
-
-  - SAMPLE_APP_URL: https://github.com/bitrise-samples/sample-apps-carthage-objc.git
+    - STEP_VERSION: 3.1.4
+    - SAMPLE_APP_URL: https://github.com/bitrise-samples/sample-apps-carthage-objc.git
+    - ORIG_BITRISE_SOURCE_DIR: $BITRISE_SOURCE_DIR
 
 workflows:
   # ----------------------------------------------------------------
   # --- workflow to Step Test
   ci:
     before_run:
-    - audit-this-step
+      - audit-this-step
+    steps:
+      - go-list:
+      - golint:
+      - errcheck:
+      - go-test:
+      - script:
+          title: Set XCODE11_12
+          inputs:
+            - content: |-
+                #!/bin/bash
+                $XCODE11_12=false
+
+                if xcodebuild -version | grep -ocE '(11|12)\..*'; then
+                  $XCODE11_12=true
+                  echo "Xcode 11 or 12 found"
+                else
+                  echo "Xcode 11 or 12 NOT found"
+                fi
+                # envman add --key XCODE11_12 --value $isXCODE11_12
+    after_run:
+      - test-without-xcconfig
+      - test-xcconfig-from-env
+      - test-xcconfig-from-input
+      - test-xcconfig-from-input-url
+      - test-cache-xcconfig-from-env
```

Comment:
Should we add tests for carthage and swift version checks too?

Reply:
Carthage version is just logged but the value is never used. 
Swift version is used in the `Cachefile`, but its value is not parsed just simply placed there. It would make more sense to test this, but I'm not sure how we could assert the output as swift version is changing on the machines. How would you test the swift version check?

Reply:
In `stepman` we do it by checking the format of the output the command prints: https://github.com/bitrise-io/stepman/blob/master/_tests/integration/version_test.go
Definitely not a must, but something to think about

Reply:
I would not get into parsing the output, that is too error prone. I think we are fine as is.